<launch>
    <arg name="manager"                 default="realsense_ros_camera_manager"/>
    <arg name="topic_camera_pose"       default="camera_pose"/>
    <arg name="topic_reloc_pose"        default="reloc_pose"/>
    <arg name="topic_pose2d"            default="pose2d" />
    <arg name="topic_map"               default="map" />
    <arg name="topic_tracking_accuracy" default="tracking_accuracy" />
    <arg name="topic_odom"              default="odom" />
    <arg name="publish_odometry"        default="false" />
    <arg name="map_resolution"          default="0.05" />
    <arg name="hoi_min"                 default="-0.5" />
    <arg name="hoi_max"                 default="0.1" />
    <arg name="doi_min"                 default="0.3" />
    <arg name="doi_max"                 default="3.0" />
  
    <node pkg="nodelet" type="nodelet" name="realsense_ros_slam" args="load realsense_ros_slam/SNodeletSlam $(arg manager)" output="screen">
        <rosparam file="$(find realsense_ros_slam)/launch/params.yaml" command="load"/>            
        <param name="topic_camera_pose"   type="str"  value="$(arg topic_camera_pose)" />           
        <param name="topic_reloc_pose"   type="str"  value="$(arg topic_reloc_pose)" />           
        <param name="topic_pose2d"   type="str"  value="$(arg topic_pose2d)" />            
        <param name="topic_map"   type="str"  value="$(arg topic_map)" />            
        <param name="topic_tracking_accuracy"   type="str"  value="$(arg topic_tracking_accuracy)" />
        <param name="topic_odom"   type="str"  value="$(arg topic_odom)" />
        <param name="publish_odometry"   type="bool"  value="$(arg publish_odometry)" />
        <param name="map_resolution"   type="double"  value="$(arg map_resolution)" />
        <param name="hoi_min"   type="double"  value="$(arg hoi_min)" />
        <param name="hoi_max"   type="double"  value="$(arg hoi_max)" />
        <param name="doi_min"   type="double"  value="$(arg doi_min)" />
        <param name="doi_max"   type="double"  value="$(arg doi_max)" />
    </node>
     <!--launch-prefix="gdb -ex run --><!--args"/-->

    <remap from="topic_camera_pose"         to="$(arg topic_camera_pose)" />
    <remap from="topic_reloc_pose"          to="$(arg topic_reloc_pose)" />
    <remap from="topic_pose2d"              to="$(arg topic_pose2d)" />
    <remap from="topic_map"                 to="$(arg topic_map)" />
    <remap from="topic_tracking_accuracy"   to="$(arg topic_tracking_accuracy)" />
</launch>
